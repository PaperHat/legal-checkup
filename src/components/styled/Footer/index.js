import styled from 'styled-components'

const Footer = styled.div`
  text-align: center;
`
Footer.displayName = 'StyledFooter'

export default Footer
