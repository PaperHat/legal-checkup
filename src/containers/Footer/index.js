import {
  PRIVACY_POLICY_PATH,
  TERMS_OF_USE_PATH,
  COMMUNITY_LAW_PATH
} from '@state/constants'

import StyledDesktopFooter from '@components/styled/DesktopFooter'
import LinkExternal from '@components/styled/LinkExternal'
import FooterLink from '../FooterLink'
import React from 'react'

export default function Footer () {
  return (
    <StyledDesktopFooter>
      <LinkExternal href={COMMUNITY_LAW_PATH} target='_blank'>
        2018 by Community Law Wellington and Hutt Valley
      </LinkExternal>
      <FooterLink to={PRIVACY_POLICY_PATH}>Privacy Policy</FooterLink>
      <FooterLink to={TERMS_OF_USE_PATH}>Terms of Use</FooterLink>
    </StyledDesktopFooter>
  )
}
