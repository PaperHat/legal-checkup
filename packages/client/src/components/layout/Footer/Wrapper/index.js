import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const Footer = styled.div`
  background-color: #202020;
  border-top: 7px solid #444141;
  display: flex;
  justify-content: center;
  align-items: center;

  ${styledMap('format', {
    [DESKTOP]: `
      height: 98px;`,
    [MOBILE]: `
      height: 135px;
      flex-direction: column;`,
    [TABLET]: `
      height: 100px;`
  })}
`
Footer.displayName = 'Footer'

export default Footer
