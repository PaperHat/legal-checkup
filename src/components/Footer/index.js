import React from 'react'
import { StyledFooter, StyledLink } from '../styled'

export default function Footer () {
  return (
    <StyledFooter>
      <div>
        <StyledLink href='/privacy-policy'>Privacy Policy</StyledLink>
        {` `}
        <StyledLink href='/terms-of-use'>Terms of Use</StyledLink>
      </div>
    </StyledFooter>
  )
}
