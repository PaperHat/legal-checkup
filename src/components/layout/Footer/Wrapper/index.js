import styled from 'styled-components'
import styledMap from 'styled-map'

import { DESKTOP, MOBILE, TABLET } from '../../../../constants'

const Footer = styled.div`
  position: absolute;
  background-color: #202020;
  border-top: 7px solid #444141;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  bottom: 0
    ${styledMap('format', {
    [DESKTOP]: `
      height: 98px;`,
    [TABLET]: `
      height: 100px;`,
    [MOBILE]: `
      height: 135px;
      flex-direction: column;`
  })};
`
Footer.displayName = 'Footer'

export default Footer
