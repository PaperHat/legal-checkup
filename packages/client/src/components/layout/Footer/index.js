import * as React from 'react'
import styled from 'styled-components'

import { COMMUNITY_LAW_URL, PRIVACY_POLICY_PATH, TERMS_OF_USE_PATH } from '../../../state/constants'
import Link from '../../Link'

import Wrapper from './Wrapper'

const Block = styled.div`
  display: flex;
  justify-content: center;
`

export default function Footer ({ format }) {
  return (
    <Wrapper format={format}>
      <Block>
        <Link href={COMMUNITY_LAW_URL} format={`${format}Footer`}>
          &copy; 2018 Community Law Wellington &amp; Hutt Valley
        </Link>
        <Link to={PRIVACY_POLICY_PATH} format={`${format}Footer`}>Privacy Policy</Link>
        <Link to={TERMS_OF_USE_PATH} format={`${format}Footer`}>Terms of Use</Link>
      </Block>
    </Wrapper>
  )
}
