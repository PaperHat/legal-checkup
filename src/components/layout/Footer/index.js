import * as React from 'react'

import { COMMUNITY_LAW_URL, PRIVACY_POLICY_PATH, TERMS_OF_USE_PATH } from '../../../state/constants'
import {
  footerWrapper,
  communityLawUrl,
  privacyPolicyPath,
  termsOfUsePath
} from '../../../constants'
import Link from '../../Link'
import Wrapper from './Wrapper'

export default function Footer ({ format }) {
  return (
    <Wrapper format={format} data-testid={footerWrapper}>
      <Link href={COMMUNITY_LAW_URL} format={`${format}Footer`} data-testid={communityLawUrl}>
      &copy; {new Date().getFullYear()} Community Law Wellington &amp; Hutt Valley
      </Link>
      <Link to={PRIVACY_POLICY_PATH} format={`${format}Footer`} data-testid={privacyPolicyPath}>Privacy Policy</Link>
      <Link to={TERMS_OF_USE_PATH} format={`${format}Footer`} data-testid={termsOfUsePath}>Terms of Use</Link>
    </Wrapper>
  )
}
