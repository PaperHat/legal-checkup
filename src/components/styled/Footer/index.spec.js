import React from 'react'
import { shallow } from 'enzyme'

import Footer from '.'

describe('components:styled:Footer', function () {
  it('renders the Footer and CSS properties properly', () =>
    expect(toJson(shallow(<Footer>Footer</Footer>))).toMatchSnapshot())
})
