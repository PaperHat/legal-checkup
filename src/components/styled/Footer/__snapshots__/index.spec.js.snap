// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:styled:Footer renders the Footer and CSS properties properly 1`] = `
.c0 {
  text-align: center;
}

<div
  className="c0"
>
  Footer
</div>
`;
